src/System.Web.OData/OData/Query/Expressions/PropertyContainer.cs

```diff
@@ -68,32 +68,39 @@ public static Expression CreatePropertyContainer(IList<NamedPropertyExpression>
             {
                 NamedPropertyExpression property = properties.First();
                 int count = properties.Count - 1;
-                int leftSize = GetLeftSize(count);
-                Expression leftNext = CreatePropertyContainer(properties.Skip(1).Take(leftSize).ToList());
-                Expression rightNext = CreatePropertyContainer(properties.Skip(1 + leftSize).ToList());
-                container = CreateNamedPropertyCreationExpression(property, leftNext, rightNext);
+                List<Expression> nextExpressions = new List<Expression>();
+                int parts = SingleExpandedPropertyTypes.Count - 1;
+                int offset = 0;
+                for (int step = parts; step > 0; step--)
```

Comment:
is there a way to re-write this iterative and not use recursion?

Reply:
It will have the same depth as final expression. And it didn't fail even without b-tree in than place. #WontFix